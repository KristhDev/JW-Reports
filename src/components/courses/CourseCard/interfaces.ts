import { Course } from '../../../interfaces/courses';

export interface CourseCardProps {
    course: Course;
    onActiveOrSuspend: () => void;
    onDelete: () => void;
    onFinishOrStart: () => void;
}