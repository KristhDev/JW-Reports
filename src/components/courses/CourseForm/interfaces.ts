export interface CourseFormValues {
    person_name: string;
    person_about: string;
    person_address: string;
    publication: string;
}