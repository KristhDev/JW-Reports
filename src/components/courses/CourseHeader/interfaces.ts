export interface RevisitHeaderProps {
    editButton: boolean;
    deleteButton: boolean;
}