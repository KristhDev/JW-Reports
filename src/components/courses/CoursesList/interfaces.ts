import { CourseFilter } from '../../../interfaces/courses';

export interface CoursesListProps {
    emptyMessage: string;
    filter: CourseFilter;
    title: string;
}

export interface ListEmptyComponentProps {
    msg: string;
}