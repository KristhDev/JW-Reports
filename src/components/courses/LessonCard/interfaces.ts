import { Lesson } from '../../../interfaces';

/* Defining the props that the LessonCard component will receive. */
export interface LessonCardProps {
    lesson: Lesson;
    screenToNavigate: string;
    onClick?: () => void;
    onDelete: () => void;
    onFinish: () => void;
}