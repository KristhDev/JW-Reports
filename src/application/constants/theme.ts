import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];