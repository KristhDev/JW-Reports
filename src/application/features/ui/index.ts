export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';