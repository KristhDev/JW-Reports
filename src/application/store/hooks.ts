import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();