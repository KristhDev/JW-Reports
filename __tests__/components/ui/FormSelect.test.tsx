import React from 'react';
import { Formik } from 'formik';
import { render, screen, fireEvent, act } from '@testing-library/react-native';

/* Components */
import { FormSelect } from '../../../src/components/ui';

/* Hooks */
import { useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Interfaces */
import { ItemOption } from '../../../src/interfaces/ui';

const itemsOptions: ItemOption[] = [
    { label: 'Test option 1', value: 'test-option-1' },
    { label: 'Test option 2', value: 'test-option-2' },
    { label: 'Test option 3', value: 'test-option-3' },
    { label: 'Test option 4', value: 'test-option-4' }
];

const labelOption = 'Opciones:';
const nameOption = 'option';
const placeholderOption = 'Seleccione una opción';
const titleOption = 'Seleccione una opción';

/* Mock hooks */
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <FormSelect /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    const renderComponent = () => {
        render(
            <Formik
                initialValues={{ option: itemsOptions[0].value }}
                onSubmit={ () => {} }
            >
                { () => (
                    <FormSelect
                        items={ itemsOptions }
                        label={ labelOption }
                        name={ nameOption }
                        placeholder={ placeholderOption }
                        title={ titleOption }
                    />
                ) }
            </Formik>
        );
    }

    it('should to match snapshot', () => {
        renderComponent();
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective props', () => {
        renderComponent();

        /* Get label of field */
        const label = screen.getByTestId('form-select-label');

        /* Check if label exists and contain text pass by props */
        expect(label).toBeTruthy();
        expect(label.props.children).toBe(labelOption);
    });

    it('should change value of select option', async () => {
        renderComponent();

        /* Get touchable */
        const touchable = screen.getByTestId('form-select-touchable');
        fireEvent.press(touchable);

        await act(async () => {

            /* Get radio touchable */
            const options = screen.getAllByTestId('radio-btn-text');
            fireEvent.press(options[1]);
        });

        /* Get text value of select option */
        const textValue = screen.getByTestId('form-select-text-value');

        /* Check if text contain option selected */
        expect(textValue.props.children).toBe(itemsOptions[1].label);
    });
});