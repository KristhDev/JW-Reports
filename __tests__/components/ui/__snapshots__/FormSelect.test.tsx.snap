// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FormSelect /> component should to match snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "display": "flex",
        "padding": 10,
        "width": 675,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 5,
            "padding": 5,
          },
          undefined,
        ]
      }
      testID="form-select-label"
    >
      Opciones:
    </Text>
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 0,
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 5,
                "flex": 1,
              }
            }
            testID="form-select-touchable"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 10,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flex": 1,
                      "fontSize": 17,
                    },
                    undefined,
                  ]
                }
                testID="form-select-text-value"
              >
                Test option 1
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#BFBFBF",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïˆ’
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />,
]
`;
